vendor/bat-native-ads/src/bat/ads/internal/ml/data/vector_data.cc

```diff
@@ -150,13 +193,45 @@ void VectorData::Normalize() {
   }
 }
 
-int VectorData::GetDimensionCountForTesting() const {
-  return storage_->dimension_count();
+int VectorData::GetDimensionCount() const {
+  return storage_->DimensionCount();
+}
+
+int VectorData::GetNonZeroElementsCount() const {
+  if (storage_->DimensionCount() == 0) {
+    return 0;
+  }
+
+  int non_zero_count = 0;
+  size_t v_index = 0;
+  while (v_index < storage_->GetSize()) {
+    if (storage_->values()[v_index] != 0) {
+      non_zero_count += 1;
+    }
+    ++v_index;
+  }
+  return non_zero_count;
 }
 
 const std::vector<float>& VectorData::GetValuesForTesting() const {
   return storage_->values();
 }
 
+const std::string VectorData::GetVectorAsString() const {
+  if (!storage_->DimensionCount()) {
+    return {};
+  }
+
+  std::vector<std::string> vector_as_string;
+  int v_index = 0;
+  int storage_size = storage_->GetSize();
+  while (v_index < storage_size) {
```

Comment:
if storage_size can be zero, can we please return early? If it should never be zero can we please add a `DCHECK`